internal/parse/parse.go

```diff
@@ -34,3 +36,47 @@ func Image(s string) error {
 
 	return nil
 }
+
+// DeploymentLogsForString parses the container logs of the specified ns/name
+// deployment, returning true if the string was found.
+func DeploymentLogsForString(ns, name, container, expectedString string) (bool, error) {
+	cmdPath, err := exec.LookPath("kubectl")
+	if err != nil {
+		return false, err
+	}
+	slashName := fmt.Sprintf("deployment/%s", name)
```

Comment:
Are we not doing this with client-go because of dependency things? Seems like it would probably be better to use client-go if we could, to avoid shelling out.

Reply:
@youngnick I created https://github.com/projectcontour/contour-operator/issues/200 to track transitioning from the controller-runtime client to client-go. I would like to stick with the current approach, since this PR does help prevent regressions and it will allow me to focus on service apis.